src/executor/insert.rs

```diff
@@ -139,8 +141,9 @@ where
 async fn exec_insert_with_returning<A, C>(
     primary_key: Option<ValueTuple>,
     mut insert_statement: InsertStatement,
+    enforce_return: bool,
```

Comment:
Maybe instead of doing this, `insert_or` should always call `exec` instead of `exec_with_returning`?

It would force you to always have to do two queries if you care about getting the return value of your insert though

Reply:
Executing two queries even when the db supports returning isn't ideal

Reply:
Maybe instead of adding this extra `enforce_return` parameter, we should instead return an iterator/array

The reason I think this might be better is that I think there are 4 cases we care about:
1) No return because either insert failed
2) No return because a DO NOTHING conflict was triggered
3) Single return (from an `insert` call
4) Multiple return (from an `insert_many` call

So instead of introducing a `enforce_return`, we could handle the cases as follows
1) Return a DbErr just like now
2) Return an empty iterator
3) Return an iterator with a single value (but this is maybe kind of ugly user-side)
4) Return an iterator with all values

I bring this up because after this PR is merged, I want to make a PR that allows `exec_insert_with_returning` for multiple return values so I don't want to make incompatible breaking changes (but I also don't want to over-complicate this PR which leads to it never getting merged)

Another option would be to make `insert`/`insert_or` and  `insert_many` return different implementations of `Insert` with different functions.